File: src/main/java/no/nav/veilarboppfolging/repository/entity/OppfolgingStartBegrunnelse.java
package no.nav.veilarboppfolging.repository.entity;

import no.nav.veilarboppfolging.kafka.dto.StartetBegrunnelseDTO;

public enum OppfolgingStartBegrunnelse {
    ARBEIDSSOKER_REGISTRERING,
    REAKTIVERT,
    SYKMELDT_MER_OPPFOLGING,
    ARENA_SYNC, // TODO: Fjerne etter opprydding i databasen
    ARENA_SYNC_ARBS,
    ARENA_SYNC_IARBS,
    NYTT_ARBEIDSSØKERREGISTER;

    public StartetBegrunnelseDTO toStartetBegrunnelseDTO() {
        if (this == ARBEIDSSOKER_REGISTRERING || this == ARENA_SYNC_ARBS || this == NYTT_ARBEIDSSØKERREGISTER) {
            return StartetBegrunnelseDTO.ARBEIDSSOKER;
        } else { // Reativer er sykmeldt fordi arbeidsøkere automatisk er under oppfølging
            return StartetBegrunnelseDTO.SYKEMELDT_MER_OPPFOLGING;
        }
    }
}

File: src/main/kotlin/no/nav/veilarboppfolging/oppfolgingsbruker/OppfolgingStartBegrunnelse.kt
package no.nav.veilarboppfolging.oppfolgingsbruker

import no.nav.veilarboppfolging.kafka.dto.StartetBegrunnelseDTO

enum class OppfolgingStartBegrunnelse {
    ARBEIDSSOKER_REGISTRERING,
    ARENA_SYNC,  // TODO: Fjerne etter opprydding i databasen
    ARENA_SYNC_ARBS,
    ARENA_SYNC_IARBS,
    MANUELL_REGISTRERING_VEILEDER;

    fun toStartetBegrunnelseDTO(): StartetBegrunnelseDTO {
        if (this == ARBEIDSSOKER_REGISTRERING || this == ARENA_SYNC_ARBS) {
            return StartetBegrunnelseDTO.ARBEIDSSOKER
        } else { // Reativer er sykmeldt fordi arbeidsøkere automatisk er under oppfølging
            return StartetBegrunnelseDTO.SYKEMELDT_MER_OPPFOLGING
        }
    }
}
